Storage/SQL/SQLiteHistoryRecommendations.swift

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+import XCGLogger
+import Deferred
+
+/// Small enum to help strength parameter requirements for potential bookmark highlights query below.
+private enum BookmarkSource {
+    case Mirror
+    case Local
+
+    var modifiedColumn: String  {
+        switch self {
+        case Mirror: return "server_modified"
+        case Local: return "local_modified"
+        }
+    }
+
+    var tableName: String {
+        switch self {
+        case Mirror: return TableBookmarksMirror
+        case Local: return TableBookmarksLocal
+        }
+    }
+}
+
+extension SQLiteHistory: HistoryRecommendations {
+    public func getHighlights() -> Deferred<Maybe<Cursor<Site>>> {
+        let limit = 19
+        let microsecondsPerMinute: Timestamp = 60_000_000 // 1000 * 1000 * 60
```

Comment:
Technically this isn't a `Timestamp` — [Alan Eliasen would murder you](http://futureboy.us/frinkdata/units.txt). It's a factor.

Just use `UInt64`.